src/Kriti.hs

```diff
@@ -38,6 +39,12 @@ runKriti templateSrc source = do
   template' <- first KritiParseError $ parser templateSrc'
   first KritiEvalError $ runEval templateSrc' template' source
 
+runKritiWith :: T.Text -> [(T.Text, J.Value)] -> [(T.Text, J.Value -> Either EvalError J.Value)] -> Either KritiError J.Value
```

Comment:
i think we actually want our functions to be `ValueExt -> ExceptT EvalError (Reader Ctxt) ValueExt` so that we can easily run them in `eval` with access to the json context.

Reply:
Using `ValueExt` instead of `Aeson.Value` will force users to either use the functions that can take primitive types as input (types that don't need evaluation) or users would have to evaluate `ValueExt` on their own, which is not intuitive for noob users.
i.e. the user can do the following:
`{{myFunction "my function's input"}}`
but the following would need to be evaluated by the user themself:
`{{myFunction $.someKey}}`